fastlane_core/lib/fastlane_core/device_manager.rb

```diff
@@ -7,13 +7,25 @@
 module FastlaneCore
   class DeviceManager
     class << self
+      attr_reader :runtime_build_os_versions
       def all(requested_os_type = "")
         return connected_devices(requested_os_type) + simulators(requested_os_type)
       end
 
       def simulators(requested_os_type = "")
         UI.verbose("Fetching available simulator devices")
 
+        output, status = Open3.capture2('xcrun simctl list runtimes -j')
+        begin
+          raise status unless status.success?
+          json = JSON.parse(output)
+          @runtime_build_os_versions = json['runtimes'].map { |h| [h['buildversion'], h['version']] }.to_h
+        rescue StandardError => e
+          UI.error(e)
+          UI.error('xcrun simctl CLI broken; cun `xcrun simctl list runtimes` and make sure it works')
+          UI.user_error!('xcrun simctl not working')
+        end
+
```

Comment:
Now that everything is cleaned up I'm wondering if here is the best place to fill this hash, especially since it's not used by the DeviceManager class itself. So that means that having that filled only in `simulators(…)` means it's dependent on the caller / user of DeviceManager to call things in a specific order.

What do you think about moving this initialization code to a memoized getter instead?

```ruby
def runtime_build_os_version
  @runtime_build_os_version ||= begin
    … logic from these lines above
  end
end
```

Reply:
This also simplified many unit tests :)